apps/cowswap-frontend/src/common/containers/TradeApprove/useTradeApproveCallback.ts

```diff
@@ -54,7 +54,7 @@ export function useTradeApproveCallback(amountToApprove?: CurrencyAmount<Currenc
             updateTradeApproveState({ error: typeof error === 'string' ? error : error.message || error.toString() })
           }
 
-          throw error
+          return undefined
```

Comment:
The only one I could fix on our side

Reply:
Shouldn't we return undefined for the handled cases? and for the unhandled, we still throw?

Reply:
All of them are handled
<img width="976" alt="image" src="https://github.com/cowprotocol/cowswap/assets/43217/e8c2d74a-09dc-4e65-b86d-e0f531f29809">

Reply:
I'm not sure I would consider the else of `if (isRejectRequestProviderError(error))` handled. Yes, shows something to the user because it updates the state, but its a non-expected error. Not part f the normal flow

shouldn't we re-throw in this case and add the original error as the cause, so sentry can give more details?